app/services/api/record_representation_order.rb

```diff
@@ -42,10 +44,11 @@ def request_body
         applicationReference: application_reference,
         statusDate: status_date,
         effectiveStartDate: effective_start_date,
+        effectiveEndDate: effective_end_date,
         defenceOrganisation: defence_organisation
-      }
+      }.compact
```

Comment:
just a question: is for request to CP and isis this a general approach you are adopting - to not send a key if it is `nil` value?

Reply:
Yep, the general approach is to not send the key for `nil` values, since the presence of optional values expects them to validate against the schema. 
More specifically, in this case `effective_end_date` sent across to HMCTS is optional, but if present, should be a date. https://github.com/ministryofjustice/hmcts-common-platform-mock-api/blob/0bd1cb06934abe89012c477fcb67c9167a0e1c91/lib/schemas/api/progression.recordRepresentationOrder.json#L35-L38